docs/src/modeler/time_series.md

```diff
@@ -0,0 +1,293 @@
+look-ahead window.
+- `Scenarios`: Stores a set of probable trajectories for forecasted quantity
+with equal probability.
+
+## Storage
+By default PowerSystems stores time series data in an HDF5 file. This prevents
+large datasets from overwhelming system memory. If you know that your dataset
+will fit in your computer's memory then you can increase performance by storing
+it in memory. Here is an example of how to do this:
+
+```julia
+sys = System(100.0; time_series_in_memory = true)
+```
+
+## Creating Time Series Data
+PowerSystems supports several methods to load time series data (Forecasts or
+StaticTimeSeries) into a System.
+- Automated parsing during system construction.  Refer to the
+[parsing documentation](parsing.md).
+- Create from TimeSeries.TimeArray or DataFrames.DataFrame
+```julia
+    resolution = Dates.Hour(1)
+    dates = range(DateTime("2020-01-01T00:00:00"), step = resolution, length = 24)
+    data = TimeArray(dates, ones(24))
+    ts = SingleTimeSeries("max_active_power", data)
+```
+```julia
+    resolution = Dates.Hour(1)
+    data = Dict(
+        DateTime("2020-01-01T00:00:00") => ones(24),
+        DateTime("2020-01-01T01:00:00") => ones(24),
+    )
+    forecast = Deterministic("max_active_power", data, resolution)
+```
+- Load from CSV file. For Deterministic forecasts, each row represents one
+look-ahead window. The first column must be the initial time and the rest must
+be the forecast values. The CSV file must have no header in the first row.
+
+```julia
+    resolution = Dates.Hour(1)
+    forecast = Deterministic("max_active_power", csv_filename, component, resolution)
+```
+
+## Scaling factors
+Time series data can store actual component values (for instance MW) or scaling
+factors (0 - 1 values). By default PowerSystems treats the values in the time
+series data as physical units. In order to specify them as scaling factors, you
+must pass the accessor function that provides the multiplier value. This value
+must be passed into the forecast when you create it.
+
+Example:
+
+```julia
+    resolution = Dates.Hour(1)
+    data = Dict(
+        DateTime("2020-01-01T00:00:00") => ones(24),
+        DateTime("2020-01-01T01:00:00") => ones(24),
+    )
+    forecast = Deterministic(
+	"max_active_power",
+	data,
+	resolution,
+	scaling_factor_multiplier = get_max_active_power,
+    )
+```
+
+In this example, the forecasted component is a generator. Whenever the user
+retrieves the forecast data PowerSystems will call
+`get_max_active_power(component)` and multiply the result with the forecast
+values (scaling factors). For instance it the maximum active power returns the
+value 50.0 and the scaling factor at some time point is 0.65, the forecast
+value will correspond to 32.5.
+
+## Adding time series to the System
+Adding time series data to a system requires a component that is already
+attached to the system. Extending the example above:
+
+```julia
+    add_time_series!(sys, component, forecast)
```

Comment:
It seems like most of the example code blocks will run if you copy and paste them in the REPL, IF you've created `component`. Does it make sense to do that so that someone could follow along?

Reply:
Yes.  We could have a section at the top that creates an example system with multiple components.